Source/Core/DolphinQt2/Debugger/RegisterColumn.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/Debugger/RegisterColumn.h"
+
+#include <QMessageBox>
+
+RegisterColumn::RegisterColumn(RegisterType type, std::function<u64()> get,
+                               std::function<void(u64)> set)
+    : m_type(type), m_get_register(get), m_set_register(set)
+{
+  RefreshValue();
+  Update();
+
+  setFlags(set == nullptr ? flags() ^ Qt::ItemIsEditable : flags());
+  setData(DATA_TYPE, static_cast<quint32>(type));
+}
+
+RegisterDisplay RegisterColumn::GetDisplay()
+{
+  return m_display;
+}
+
+void RegisterColumn::SetDisplay(RegisterDisplay display)
+{
+  m_display = display;
+  Update();
+}
+
+void RegisterColumn::RefreshValue()
+{
+  QBrush brush = QPalette().brush(QPalette::Text);
+
+  if (m_value != m_get_register())
+  {
+    m_value = m_get_register();
+    brush.setColor(Qt::red);
+  }
+
+  setForeground(brush);
+
+  Update();
+}
+
+void RegisterColumn::SetValue()
+{
+  u64 value = 0;
+
+  bool valid = false;
+
+  switch (m_display)
+  {
+  case RegisterDisplay::Hex:
+    value = text().toULongLong(&valid, 16);
+    break;
+  case RegisterDisplay::SInt32:
+    value = text().toInt(&valid);
+    break;
+  case RegisterDisplay::UInt32:
+    value = text().toUInt(&valid);
+    break;
+  case RegisterDisplay::Float:
+  {
+    float f = text().toFloat(&valid);
+    value = *reinterpret_cast<u32*>(&f);
```

Comment:
This is undefined behavior. The proper way to do this is via memcpy as:

`std::memcpy(&value, &f, sizeof(u32));`

`<cstring>` also needs to be included in this case.